test/unit/spec/narrative_core/kbaseNarrativeOutlinePanel-spec.js

```diff
@@ -0,0 +1,131 @@
+define(['jquery', 'kbaseNarrativeOutlinePanel', 'base/js/namespace'], ($, Widget, Jupyter) => {
+    'use strict';
+    describe('The kbaseNarrativeOutlinePanel widget', () => {
+        let widget, $target;
+        beforeAll(() => {
+            if (!Jupyter) Jupyter = {};
+            Jupyter.notebook = { get_cells: () => [] };
+            $target = $(`<div id="test-div">`);
+            widget = new Widget($target, {
+                widget: 'kbaseNarrativeOutlinePanel',
+            });
+        });
+
+        it('should have a body', () => {
+            expect(widget.body).toBeDefined();
+        });
+
+        it('should render basic outline', () => {
+            Jupyter.notebook = {
+                get_cells: () => [
+                    mockCell('markdown', null, false, 'test-icon', '<h1>test header!!</h1>'),
+                    mockCell('code', 'My App', false),
```

Comment:
Check the `test/unit/mocks` module -- we already have quite a few mocks for mocking the notebook, cells, etc. Ideally all tests would use those mocks so that we don't have to go back and edit individual tests if there are changes to the cell structure or notebook functionality. If the current mock cell doesn't work for your purposes, feel free to edit it so that it does.

Reply:
Ah! Ill check it out

Reply:
It looks like these are quite different, as what I'm mostly mocking is the rendered portion of a cell, vs it's attributes. I think wrapping the existing mock is probably better... I'll take a look